### PR TITLE
Release 2.10.0

### PR DESCRIPTION
**Pull request by LukeHandle**
Supercedes https://github.com/minotar/imgd/pull/182

 * "Fix" status by adding a 1 millisecond for each channel update test
 * Add Prometheus metrics (closing out https://github.com/minotar/imgd/issues/140 - maybe can be opened up one day)
 * Perform UUID lookups in background (fixes https://github.com/minotar/imgd/issues/181)
 * Configure Wercker to draft a release and upload binaries when launching "release" pipeline (admin only)

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -191,32 +206,50 @@ func (router *Router) Bind() {
...
+		hasTimer.ObserveDuration()
+		pullTimer := prometheus.NewTimer(cacheDuration.WithLabelValues("pull"))
+		defer pullTimer.ObserveDuration()
 		stats.HitCache()
 		return &mcSkin{Processed: nil, Skin: cache.pull(strings.ToLower(username))}
 	}
+	hasTimer.ObserveDuration()
 
-	skin, err := minecraft.FetchSkinFromMojang(username)
+	uuid, err := minecraft.NormalizePlayerForUUID(username)
```
Comment (LukeHandle) [inline http.go]: *this is rate limiting on an empty cache when live*

```
[20:45:17.399568] DEBU Failed UUID lookup: Notch (unable to GetAPIProfile: rate limited)
```

A good few minutes later and still blocked:

```
$ curl https://api.mojang.com/users/profiles/minecraft/LukeHandle
{"error":"TooManyRequestsException","errorMessage":"The client has sent too many requests within a certain amount of time"}
```

Either the initial herd of traffic is too large when trying an empty cache, or I'm missing some caching layer somewhere here. This should only be called when not in the Redis cache, so it's not exactly being hit 24/7. But it is late 😴

**Comment by LukeHandle** [comment]
@connor4312 If you get a moment to took at this then awesome. I tried deploying live and it initially was okay. I then cleared Redis as I realised we would have a fair amount of stale data there - mistake. Started rate limiting at this stage "unable to GetAPIProfile: rate limited".

This is how we are resolving username -> UUID:
https://github.com/minotar/minecraft/blob/958def62ff03879f87a702eaac1a7b6731de0b44/profiles.go#L32

I can see other issues here where we're not preventing simultaneous requests for the same user, but I had not considered this until about 30 minutes ago.

**Comment by LukeHandle** [comment]
I don't see this implementation working here. I just tried a moment ago (quietest part of the day-ish) and within 3 minutes we were rate-limiting on UUID lookups. We are only performing them when we have nothing in the cache, so they are hardly avoidable.

One potential idea is that we could try and use the multi Username lookup and have some sort of bucket system for requesting:

http://wiki.vg/Mojang_API#Playernames_-.3E_UUIDs

eg. we have a bucket which can fill up to 100 and every 0.5 seconds (120 req/min 🤷‍♂️ ) we send all the UUIDs in the buckets to Mojang. Any request for an uncached Username lookup will be held until the bucket submits, so will add up to 0.5 seconds to requests...

It sounds bad on paper, and even worse for me to try and implement, but I'm otherwise out of any ideas and can't see a way forward here.

**Comment by TNTUP** [comment]
Seems like Mojang wants to break off skin websites? Because of this shitty ratelimiting they have we cannot serve player skins as usual :/ it even broke the legacy Minotar PHP that was on skins.minecraft.net :/

**Pull request merged**